spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraJoinRDD.scala

```diff
@@ -47,6 +49,11 @@ class CassandraJoinRDD[L, R] private[connector](
 
   override protected val classTag = rightClassTag
 
+  override protected[connector] lazy val rowReader: RowReader[R] = manualRowReader match {
+    case Some(rowReader) => rowReader
```

Comment:
I think that @pkolaczk meant that you used the same name for lazy val and for unapply - you should just change 
``` case Some(rowReader) => rowReader ```
to something like
``` case Some(someRowReader) => someRowReader ```